tools/lxdclient/client.go

```diff
@@ -262,6 +264,12 @@ func newRawClient(remote Remote) (*lxd.Client, error) {
 		return nil, errors.Trace(err)
 	}
 
+	// Replace the proxy handler with the one managed
+	// by Juju's worker/proxyupdater.
+	if tr, ok := client.Http.Transport.(*http.Transport); ok && tr.Proxy != nil {
```

Comment:
While this is a bit ugly, if we're going to do it, shouldn't we be universally setting tr.Proxy and not just setting it if there is already a proxy? (from what I can tell, its probably always setting ProxyFromEnvironment, which also does the read-only-once stuff.)